invenio_users_resources/services/users/config.py

```diff
@@ -47,7 +47,7 @@ class UserSearchOptions(SearchOptions, SearchOptionsMixin):
     query_parser_cls = QueryParser.factory(
         tree_transformer_cls=SearchFieldTransformer,
         fields=["username^2", "email^2", "profile.full_name^3", "profile.affiliations"],
-        allow_list=["username", "email"],  # mapped fields are added on the query parser
+        allow_list=["username", "email", "id"],  # mapped fields are added on the query parser
```

Comment:
I have some doubts if we should allow to search by ID, I think it enables it for everyone, not only admins

Reply:
I tested and it's independent of the visibility set in the profile, if visibility is set to hidden, then no one can find users by id. If is set to Public, users can be found by email, username and now ID as well